src/index.ts

```diff
@@ -0,0 +1,178 @@
+import { SecretManagerServiceClient, protos } from "@google-cloud/secret-manager";
+
+/**
+ * Definition of filter function
+ */
+type FilterFunction = (name: string) => boolean;
+
+/**
+ * Definition of possible options for the module
+ */
+type WithGoogleSecretsOptions = {
+  /**
+   *
+   */
+  projectName: string;
+  /**
+   *
+   */
+  mapping: Record<string, string | string[]>;
+  /**
+   *
+   */
+  filter?: string | FilterFunction;
+  /**
+   *
+   */
+  version?: string;
+  /**
+   *
+   */
+  nextConfig: object;
+};
+
+/**
+ * Definition of the config
+ */
+type Config = {
+  [key: string]: string | Config;
+};
+
+/**
+ * Checks if the data is a secret payload
+ * @param data the possible secret payload
+ * @returns True if it is a secret payload
+ */
+function isSecretPayload(data: Uint8Array | string | null | undefined): data is Uint8Array {
+  return data !== null;
+}
+
+/**
+ * Checks if object is config
+ * @param config the possible config object
+ * @returns True if it is a config object
+ */
+function isConfig(config: Config | string): config is Config {
+  return config !== null;
+}
+
+/**
+ * Gets the short name of a secret from the long name
+ * @param name The long name
+ * @returns The short name
+ */
+function getSecretName(name: string | undefined | null): string {
+  if (!name || !name?.length) return "";
+  const splits = name.split("/");
+  return splits[splits.length - 1];
+}
+
+/**
+ * Sets a specific config value based on the path
+ * @param config The config object
+ * @param path The path
+ * @param value The value to set
+ * @param subPath The sub path
+ */
+function setConfigurationValue(config: Config, path: string, value: string, subPath?: string): void {
+  const pathSplit = (subPath ?? path).split(/\.|__/);
+  const [currentName] = pathSplit;
+  if (pathSplit.length > 1) {
+    if (typeof config[currentName] !== "object") config[currentName] = {};
+    if (isConfig(config[currentName])) {
+      setConfigurationValue(config[currentName] as Config, path, value, pathSplit.filter((_, i) => i > 0).join("."));
+    } else {
+      console.warn("couldn't override following config:", path);
```

Comment:
Can we prefix all console outputs with something like `WithGoogleSecrets - ` 

So it's a bit clearer where it comes from.

Reply:
yes sure, prefix added